stackit/internal/services/resourcemanager/project/resource.go

```diff
@@ -421,42 +522,123 @@ func mapFields(ctx context.Context, projectResp *resourcemanager.GetProjectRespo
 	}
 	model.Name = types.StringPointerValue(projectResp.Name)
 	model.Labels = labels
+
+	err = mapMembers(membersResp, model)
+	if err != nil {
+		return fmt.Errorf("map members: %w", err)
+	}
+
+	return nil
+}
+
+func mapMembers(members *[]authorization.Member, model *Model) error {
+	if members == nil {
+		model.Members = types.ListNull(types.ObjectType{AttrTypes: memberTypes})
+		return nil
+	}
+
+	if (model.Members.IsNull() || model.Members.IsUnknown()) && !model.OwnerEmail.IsNull() {
+		// If the new "members" field is not set and the deprecated "owner_email" field is set,
+		// we keep the old behavior and do map the members to avoid an inconsistent result after apply error
+		model.Members = types.ListNull(types.ObjectType{AttrTypes: memberTypes})
+		return nil
+	}
+
+	membersList := []attr.Value{}
+	for i, m := range *members {
+		if utils.IsLegacyProjectRole(*m.Role) {
+			continue
+		}
+		membersMap := map[string]attr.Value{
+			"subject": types.StringPointerValue(m.Subject),
+			"role":    types.StringPointerValue(m.Role),
+		}
+
+		memberTF, diags := types.ObjectValue(memberTypes, membersMap)
+		if diags.HasError() {
+			return fmt.Errorf("mapping index %d: %w", i, core.DiagsToError(diags))
+		}
+
+		membersList = append(membersList, memberTF)
+	}
+
+	membersTF, diags := types.ListValue(
+		types.ObjectType{AttrTypes: memberTypes},
+		membersList,
+	)
+	if diags.HasError() {
+		return core.DiagsToError(diags)
+	}
+
+	model.Members = membersTF
 	return nil
 }
 
-func toCreatePayload(model *Model, serviceAccountEmail string) (*resourcemanager.CreateProjectPayload, error) {
+func toMembersPayload(ctx context.Context, model *Model) (*[]authorization.Member, error) {
 	if model == nil {
 		return nil, fmt.Errorf("nil model")
 	}
+	if model.Members.IsNull() || model.Members.IsUnknown() {
+		return &[]authorization.Member{}, nil
+	}
 
-	owner := projectOwner
-	serviceAccountSubject := serviceAccountEmail
-	members := []resourcemanager.Member{
-		{
-			Subject: &serviceAccountSubject,
-			Role:    &owner,
-		},
```

Comment:
The service account was being silently set as an owner of the project before, but this was not documented anywhere and it would case an inconsistent result after apply in the `members` field afterwards unless we filtered it out (which would be misleading)

Reply:
I think the simplest approach is to let the users choose all members of the project themselves, as he would do in the API

Reply:
I agree